### PR TITLE
[1LP][RFR] Datastore collection + tests

### PR DESCRIPTION
**Pull request by vatsalparekh**
Purpose or Intent
=================
Collection for `cfme.infrastructure.datastore.Datastore`
{{pytest: cfme/tests/infrastructure/test_datastore_analysis.py cfme/tests/infrastructure/test_delete_infra_object.py cfme/tests/infrastructure/test_instance_analysis.py -vvvv}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit a7287d6a7ba3ee574d9842f41aab9461aece28ef:

`cfme/infrastructure/datastore.py`:
- :bangbang: Line 150:32: **E203** *whitespace before ':'*
- :red_circle: Line 259:16: **F821** *undefined name 'NoSuchElementException'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/infrastructure/datastore.py [hunk 1]
```diff
@@ -114,6 +114,63 @@ def is_displayed(self):
         return False
 
 
+class DatastoreCollection(Navigatable):
+    """Collection class for `cfme.infrastructure.datastore.Datastore`"""
+
+    def __init__(self, appliance=None):
+        appliance = appliance
```
Comment (quarckster) [inline cfme/infrastructure/datastore.py]: **REQUIRED**
```python
self.appliance = appliance
```

File: cfme/infrastructure/datastore.py [hunk 2]
```diff
@@ -114,6 +114,63 @@ def is_displayed(self):
...
+        """
+        Note:
+            Datastores must have 0 hosts and 0 VMs for this to work.
+        """
+        datastores = list(datastores)
+
+        checked_datastores = list()
+
+        view = navigate_to(self, 'All')
+
```
Comment (quarckster) [inline cfme/infrastructure/datastore.py]: **REQURIED**
Please remove these blank lines between statements.

Comment (vatsalparekh) [reply cfme/infrastructure/datastore.py]: Fixed

File: cfme/infrastructure/datastore.py (hunk 1 above)
Comment (vatsalparekh) [reply cfme/infrastructure/datastore.py]: Fixed

File: cfme/infrastructure/datastore.py [hunk 3]
```diff
@@ -114,6 +114,61 @@ def is_displayed(self):
...
+                view.entities.get_entity(by_name=datastore.name, surf_pages=True).check()
+                checked_datastores.append(datastore)
+            except ItemNotFound:
+                raise ValueError('Could not find some datastore in the UI')
+
+        if set(datastores) == set(checked_datastores):
+            view.toolbar.configuration.item_select('Perform SmartState Analysis', handle_alert=True)
+            for datastore in datastores:
+                view.flash.assert_success_message(('"{}": scan successfully '
+                                                   'initiated'.format(datastore.name)))
```
Comment (mfalesni) [inline cfme/infrastructure/datastore.py]: Missing else statement in here. If the sets aren't equal, then it should probably do something as well, probably yell with an error.

Comment (vatsalparekh) [reply cfme/infrastructure/datastore.py]: added.

**Pull request merged**